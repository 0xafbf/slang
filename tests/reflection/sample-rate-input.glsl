//TEST(smoke):SIMPLE:-profile ps_4_0 -no-checking -target reflection-json

// Check that we report sample-rate entry point input correctly

uniform texture2D t;
uniform sampler s;

sample in vec2 uv;

out vec4 c;

void main()
{
    c = texture(sampler2D(t,s), uv);
}
