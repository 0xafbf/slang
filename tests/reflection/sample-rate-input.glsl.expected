result code = 0
standard error = {
}
standard output = {
{
    "parameters": [
        {
            "name": "t",
            "binding": {"kind": "descriptorTableSlot", "index": 0},
            "type": {
                "kind": "resource",
                "baseShape": "texture2D"
            }
        },
        {
            "name": "s",
            "binding": {"kind": "descriptorTableSlot", "index": 1},
            "type": {
                "kind": "samplerState"
            }
        },
        {
            "name": "uv",
            "binding": {"kind": "vertexInput", "index": 0},
            "type": {
                "kind": "vector",
                "elementCount": 2,
                "elementType": {
                    "kind": "scalar",
                    "scalarType": "float32"
                }
            }
        },
        {
            "name": "c",
            "binding": {"kind": "fragmentOutput", "index": 0},
            "type": {
                "kind": "vector",
                "elementCount": 4,
                "elementType": {
                    "kind": "scalar",
                    "scalarType": "float32"
                }
            }
        }
    ],
    "entryPoints": [
        {
            "name": "main",
            "stage:": "fragment",
            "usesAnySampleRateInput": true
        }
    ]
}
}
